catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/RefreshRegistrySubscriptions.java

```diff
@@ -114,13 +128,18 @@ private void addOrUpdateMetacardsForRemoteRegistries(Set<String> sourceIds)
     private Map<String, Metacard> getRegistryMetacardsMap() throws FederationAdminException {
 
         Map<String, Metacard> registryMetacardsMap = new HashMap<>();
-        List<Metacard> registryMetacards = federationAdminService.getRegistryMetacards();
+        try {
+            List<Metacard> registryMetacards = javax.security.auth.Subject.doAs(subject,
+                    (PrivilegedExceptionAction<List<Metacard>>) () -> federationAdminService.getRegistryMetacards());
 
-        for (Metacard metacard : registryMetacards) {
-            String registryId = metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
-                    .getValue()
-                    .toString();
-            registryMetacardsMap.put(registryId, metacard);
+            for (Metacard metacard : registryMetacards) {
+                String registryId = metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
```

Comment:
Do we need to check if `metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)` returns null? Seems like the assumption is that metacard will have that attribute populated since it is getting the list from the `federationAdminService`. Might get flagged as a potential NPE.